addons/mail/static/src/models/message/message.js

```diff
@@ -242,7 +244,12 @@ function factory(dependencies) {
          * that Discuss and Inbox are already opened.
          */
         replyTo() {
-            this.messaging.discuss.replyToMessage(this);
+            const { discuss } = this.messaging;
+            if (discuss.replyingToMessage === this) {
+                discuss.clearReplyingToMessage();
+            } else {
+                discuss.replyToMessage(this);
+            }
```

Comment:
You asked me to move most of the action logic in models, this used to be handled by the message component and would now be handled by the actionlist component if not in models, this is the place where it made the most sense to me. We could also move it to the discuss model, but that seems like it may break stuff because its method currently override its state regardless of its previous state (the checks are all done outside of it at the moment), so it may have unforeseen consequences.

Reply:
For me this should be on discuss model, it makes no sense on message if discuss is not there.

Then you don't even have the issue about the 2 different events I was talking about earlier, it's just one event in discuss period. Just need to store the ref to the reply button in the message, and discuss can do the check of tarbet through relation.